src/Guard.php

```diff
@@ -215,131 +205,126 @@ public function generateToken()
 
         return $this->keyPair;
     }
-    
+
     /**
-     * Generates a new CSRF token and attaches it to the Request Object
-     * 
-     * @param  ServerRequestInterface $request PSR7 response object.
-     * 
-     * @return ServerRequestInterface PSR7 response object.
+     * Validate CSRF token from current request against token value stored in $_SESSION
+     *
+     * @param  string|null $name  CSRF name
+     * @param  string|null $value CSRF token value
+     *
+     * @return bool
      */
-    public function generateNewToken(ServerRequestInterface $request)
+    public function validateToken(?string $name, ?string $value): bool
```

Comment:
The two params should not be `null`. We should force them to be `string`. If so, then we need to check for `null` in the `\Slim\Csrf\Guard::process` method before calling `validateToken`.

This would also solve the fact that `hash_equals` expects a `string` as second param. Not a `null`.

Reply:
that's true, let me fix that